### PR TITLE
removing deprecated methods + change unaligned joints with boost fusion fix

### PR DESCRIPTION
**Pull request by fvalenza**
Related to #311

### CODE REVIEW

File: src/multibody/joint/joint-revolute-unaligned.hpp [hunk 1]
```diff
@@ -342,12 +339,8 @@ namespace se3
...
     {
       const double & q = qs[idx_q()];
-
-      /* It should not be necessary to copy axis in jdata, however a current bug
-       * in the fusion visitor prevents a proper access to jmodel::axis. A
-       * by-pass is to access to a copy of it in jdata. */
-      data.angleaxis.angle() = q;
-      data.M.rotation(data.angleaxis.toRotationMatrix());
+      
+      data.M.rotation(Eigen::AngleAxisd(q, axis).toRotationMatrix());
```
Comment (fvalenza) [inline src/multibody/joint/joint-revolute-unaligned.hpp]: Is it ok to create an angleaxis each time ?

**Comment by nmansard** [comment]
Less efficient.

File: src/multibody/joint/joint-revolute-unaligned.hpp (hunk 1 above)
Comment (nmansard) [inline src/multibody/joint/joint-revolute-unaligned.hpp]: I see no reasons not to keep an initialized AngleAxis in JData. 
Possibly add an assert to double check that angleaxis.axis == jmodel.axis (I guess not an isApprox but an exact ==).

File: src/multibody/model.hpp [hunk 2]
```diff
@@ -292,43 +292,8 @@ namespace se3
...
     bool existFrame (const std::string & name, const FrameType& type = (FrameType) (JOINT | FIXED_JOINT | BODY | OP_FRAME | SENSOR )) const;
     
-    ///
-    /// \brief Get the name of a frame given by its index.
-    ///
-    /// \param[in] index Index of the frame.
-    ///
-    /// \return The name of the frame.
-    ///
-    PINOCCHIO_DEPRECATED const std::string & getFrameName (const FrameIndex index) const;
```
Comment (nmansard) [inline src/multibody/model.hpp]: I think it is relevant to remove that. It is a quite recent feature, and nobody should complain.

File: src/parsers/urdf.hpp [hunk 3]
```diff
@@ -58,21 +58,6 @@ namespace se3
...
-    ///
-    /// \brief Build the model from a URDF file with a particular joint as root of the model tree.
-    ///
-    /// \param[in] filemane The URDF complete file path.
-    /// \param[in] rootJoint The joint at the root of the model tree.
-    /// \param[in] verbose Print parsing info.
-    ///
-    /// \return The se3::Model of the URDF file.
-    ///
-    PINOCCHIO_DEPRECATED
```
Comment (nmansard) [inline src/parsers/urdf.hpp]: Maybe here we should keep the deprecated method until v1.3. I am not sure. We are not very rigorous with that. I certainly will not complain.

**Comment by jmirabel** [comment]
> Less efficient.

Really ? I would have said the current implementation is at least as efficient...
On one side, you marginally increase the memory footprint. On the other, you have (not even sure) a marginally bigger call stack... With good compilation options, I suspect the temporary `Eigen::AngleAxisd` is not even created.

Did somebody (@jcarpent) run some benchmark for such things ?

File: src/multibody/joint/joint-revolute-unaligned.hpp (hunk 1 above)
Comment (nmansard) [inline src/multibody/joint/joint-revolute-unaligned.hpp]: Ok, let's keep it as you proposed, with a temporary.

**Pull request merged**